llvm/docs/LangRef.rst

```diff
@@ -12472,9 +12472,8 @@ instruction's return value on the same edge).
 The optional ``fast-math-flags`` marker indicates that the phi has one
 or more :ref:`fast-math-flags <fastmath>`. These are optimization hints
 to enable otherwise unsafe floating-point optimizations. Fast-math-flags
-are only valid for phis that return a floating-point scalar or vector
-type, or an array (nested to any depth) of floating-point scalar or vector
-types.
+are only valid for phis that return a floating-point scalar or vector type,
+possibly within an array (nested to any depth), or a homogeneous struct literal.
```

Comment:
Remove "possibly", and elaborate on what a homogenous struct means (also it's not a literal?)

Reply:
Should be "homogeneous literal struct". "literal" here means "unnamed".

Reply:
> Remove "possibly"

I think removing that makes it sound like the float/vector type is required to be within an array or struct type.

Reply:
Just list the cases that work